File: AEProblems/nonConstructibleChange.js
function nonConstructibleChange(coins) {
    coins = coins.sort((a, b) => a - b);
    let minChange = 0;

    for (let coin of coins) {
        if (coin > minChange + 1) {
            return minChange + 1;
        } else {
            minChange += coin;
        }
    }
    return minChange + 1;
}

console.log(nonConstructibleChange([5, 7, 1, 1, 2, 3, 22]))

File: AEProblems/non_constructible_change.py
def nonConstructibleChange(coins):
    # Write your code here.
    if len(coins) < 1:
        return 1

    coins = sorted(coins)
    min_change = 0

    for coin in coins:
        if coin > min_change + 1:
            return min_change + 1
        else:
            min_change += coin
    return min_change + 1


print(nonConstructibleChange([5, 7, 1, 1, 2, 3, 22]))
